0001-typedb/02-dev/03-match.md

```diff
@@ -0,0 +1,692 @@
+| owners of attribute type                | `$<var_label> owns <attr-type>`                         |
+| attribute subtype                       | `$<var> sub[!] <attribute-type> [, value <value-type>]` |
+
+### Data
+
+The patterns below can be used to find data in a database.
+
+| **What we are looking for in data** | **Pattern syntax**                                                                                                             |
+|-------------------------------------|--------------------------------------------------------------------------------------------------------------------------------|
+| instance                            | `$<var_label> isa[!] <type> [, <has-attribute expression>...]`                                                                 |
+| attribute                           | `$<var_label> [isa[!] <attr-type>] (contains "<text>" &#124; like "<regex>" &#124; <value>) [, <has-attribute expression>...]` |
+| relation                            | `[$<var_label>] ([<role>:] $<var_label> [, [<role>:] $<var_label>]...) isa <relation-type> [, <has-attribute expression>...]`  |
+| has-attribute expression            | `has <attr-type> ($<var_label> &#124; [<comparison-operator>] <value> )`                                                       |
+| comparison                          | `$<var_label> [<comparison-operator>] <value>`                                                                                 |
+| equal                               | `$<var_label> is $<var_label>`                                                                                                 |
+
+### Comparison operators
+
+The following operators are supported for comparing attribute values: `=`, `!=`, `>`, `>=`, `<` and `<=`.
+
+### Combining patterns
+
+![Combining statements](../../images/query/pattern-structure.png)
+
+By arranging statements together, we can express more complex pattern scenarios and their corresponding data.
+
+- Statement: simplest possible arrangement — a single basic building block as [explained above](#patterns-overview).
+- Conjunction: a set of patterns where to satisfy a match, all patterns must be matched. We use conjunctions by default 
+  just by separating the partaking patterns with semicolons `;`.
+- Disjunction: a set of patterns where to satisfy a match, at least one pattern must be matched. We form
+  disjunctions by enclosing the partaking patterns within curly braces {} and interleaving them with the or keyword.
+- Negation: defines a conjunctive pattern that explicitly defines conditions not to be met. We form negations by 
+  defining the pattern of interest inside a not {}; block.
+
+## Match clause
+
+### Schema queries
+
+A `match` clause can be used to find types or roles in a database schema.
+
+#### Type matching
+
+##### All types
+
+Use a `subtype` pattern with type `thing` to find all types defined in a schema.
```

Comment:
This will change in 3.0, so we might want to highlight that somehow.

Reply:
NVM, I have found a workaround:
```
match $t sub $a;
```